plugin/storage/cassandra/spanstore/query_builder.go

```diff
@@ -1,177 +0,0 @@
-
-	tagQuery = `SELECT trace_id, span_id FROM tag_index WHERE service_name = ? AND tag_key = ? AND tag_value = ?`
-)
-
-var (
-	// ErrServiceNameNotSet occurs when attempting to query with an empty service name
-	ErrServiceNameNotSet = errors.New("Service Name must be set")
-
-	// ErrStartTimeMinGreaterThanMax occurs when start time min is above start time max
-	ErrStartTimeMinGreaterThanMax = errors.New("Start Time Minimum is above Maximum")
-
-	// ErrDurationMinGreaterThanMax occurs when duration min is above duration max
-	ErrDurationMinGreaterThanMax = errors.New("Duration Minimum is above Maximum")
-
-	// ErrNumTracesNotSet occurs when num traces is not set
-	ErrNumTracesNotSet = errors.New("Num Traces must be set")
-
-	// ErrMalformedRequestObject occurs when a request object is nil
-	ErrMalformedRequestObject = errors.New("Malformed request object")
-
-	dawnOfTime = time.Unix(0, 0)
-)
-
-// Query encapsulates a query string with placeholders and parameters to bind to those placeholders
-type Query struct {
-	buffer         bytes.Buffer
-	Parameters     []interface{}
-	hasBaseClause  bool
-	hasWhereClause bool
-}
-
-// QueryString returns the final query string.
-func (q *Query) QueryString() string {
-	return q.buffer.String()
-}
-
-// append query adds some portion of a query string to a Query
-//
-// It is stateful and works in a state-machine fashion. The first call to append() just adds
-// a string and parameters, the second call adds " WHERE " plus a clause, and
-// further calls add " AND " plus another clause.
-//
-// append should be used to add anything to a Query to avoid breaking the state machine
-func (q *Query) append(queryString string, parameters ...interface{}) {
-	if !q.hasBaseClause {
-		q.hasBaseClause = true
-	} else if !q.hasWhereClause {
-		q.buffer.WriteString(" WHERE ")
-		q.hasWhereClause = true
-	} else {
-		q.buffer.WriteString(" AND ")
-	}
-
-	q.buffer.WriteString(queryString)
-	q.Parameters = append(q.Parameters, parameters...)
-}
-
-func (q *Query) appendNotEmpty(queryString string, value string) {
-	if value != "" {
-		q.append(queryString, value)
-	}
-}
-
-func (q *Query) appendTimeIfNotZero(queryString string, value time.Time) {
-	if !value.IsZero() {
-		microseconds := value.Sub(dawnOfTime).Nanoseconds() / 1000
-		q.append(queryString, microseconds)
-	}
-}
-
-func (q *Query) appendDurationIfNotZero(queryString string, value time.Duration) {
-	if value != 0 {
-		microseconds := uint64(value / 1000)
-		q.append(queryString, microseconds)
-	}
-}
-
-func (q *Query) appendLimit(value int) {
-	q.buffer.WriteString(limitClause)
-	q.Parameters = append(q.Parameters, value)
-}
-
-// Queries is a construct that provides the main query against cassandra and any optional tag queries
-type Queries struct {
-	spanstore.TraceQueryParameters
-	MainQuery  Query
-	TagQueries []Query
-}
-
-// BuildQueries takes a query request and generates Queries
-func BuildQueries(p *spanstore.TraceQueryParameters) (*Queries, error) {
```

Comment:
I can concede on that because query logic is more complicated (esp. duration), it make sense to have dedicated methods. I disagree that validation is unnecessary - look at the errors at the top of this file, they have nothing to do with querying logic, but with the consistency of the query parameters passed in.